core/src/main/java/org/jruby/ir/IRMethod.java

```diff
@@ -138,17 +139,33 @@ public InterpreterContext builtInterpreterContext() {
      *
      * @return appropriate interpretercontext
      */
-    public synchronized InterpreterContext builtInterperterContextForJavaConstructor() {
-        InterpreterContext interpreterContext = builtInterpreterContext();
+    public ExitableInterpreterContext builtInterperterContextForJavaConstructor() {
```

Comment:
This method name is misspelled... perhaps since this would be an API signature change, we deprecate the old spelling and add a new method with corrected spelling ("builtInterpreterContextForJavaConstructor") that has the more specific signature?

Reply:
:+1: good call, missed the spelling issue.